editor/modes/visual-editor/block.js

```diff
@@ -51,13 +51,28 @@ import {
 	getMultiSelectedBlockUids,
 } from '../../selectors';
 
-const { BACKSPACE, ESCAPE, DELETE, UP, DOWN, LEFT, RIGHT, ENTER } = keycodes;
+const { BACKSPACE, ESCAPE, DELETE, UP, DOWN, LEFT, RIGHT, ENTER, TAB } = keycodes;
+const F10 = 121;
 
 function FirstChild( { children } ) {
 	const childrenArray = Children.toArray( children );
 	return childrenArray[ 0 ] || null;
 }
 
+function selectAncestor( node, selector ) {
+	while ( node !== null ) {
+		if ( node.querySelector( selector ) !== null ) {
+			return node;
+		}
+		node = node.parentNode;
+	}
+	return null;
+}
+
+function isToolbar( target ) {
+	return selectAncestor( target, '.components-toolbar, .editor-block-settings-menu, .editor-block-mover' ) !== null;
```

Comment:
A component really ought not traverse the DOM, especially through nodes not immediately rendered by its own `render`, since it establishes dependencies between the components which are difficult to maintain and debug. It would not be very obvious to add a new entry to this list if we add another toolbar, for example.

Why do we not capture this event at the specific toolbar component(s) instead? If we then need to communicate this to another component, we can employ one of a [number of strategies](http://andrewhfarmer.com/component-communication/).

Aside: We have [`element-closest`](https://www.npmjs.com/package/element-closest) available as a polyfill for [`Element#closest`](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest).